src/UndockedRegFreeWinRT/UndockedRegFreeWinRT/catalog.cpp

```diff
@@ -193,20 +193,19 @@ HRESULT WinRTGetMetadataFile(
     {
         return E_NOTIMPL;
     }
-
-    Microsoft::WRL::Wrappers::HString substring;
-    if (SUCCEEDED(WindowsSubstringWithSpecifiedLength(name, 0, 7, substring.GetAddressOf())))
+    wil::unique_hstring substring;
+    if (SUCCEEDED(WindowsSubstringWithSpecifiedLength(name, 0, 7, &substring)))
     {
         int result;
-        if (SUCCEEDED(WindowsCompareStringOrdinal(substring.Get(), HStringReference(L"Windows").Get(), &result)) && (result == 0))
+        if (SUCCEEDED(WindowsCompareStringOrdinal(substring.get(), HStringReference(L"Windows").Get(), &result)) && (result == 0))
         {
             // Let original RoGetMetadataFile handle Windows namespace
             return REGDB_E_CLASSNOTREG;
         }
     }
 
     DWORD metaDataFilePathsCount = 0;
-    HSTRING* metaDataFilePaths;
+    wil::unique_cotaskmem_array_ptr<wil::unique_hstring> metaDataFilePaths;
```

Comment:
This solves two leaks (the array being leaked, as well as all of the elements), but now it's going to overrelease the string that's returned.

```
*metaDataFilePath = metaDataFilePaths[bestMatch];
metaDataFilePaths[bestMatch] = nullptr;
```